packages/p2p/src/components/modal-manager/modals/filter-modal/filter-modal.jsx

```diff
@@ -39,156 +89,317 @@ const FilterModal = () => {
         buy_sell_store.setSelectedPaymentMethodText([]);
     };
 
-    const onBackdropClick = () => {
+    const onClickClose = () => {
         buy_sell_store.setShowFilterPaymentMethods(false);
-        hideModal();
         my_profile_store.setSearchTerm('');
-        if (selected_methods_text.length && !buy_sell_store.selected_payment_method_value.length) {
-            onClickClear();
-        }
+        my_profile_store.setSearchResults([]);
+        hideModal({
+            should_restore_local_state: false,
+        });
     };
 
+    const diff = (arr1, arr2) => arr1.filter(x => !arr2.includes(x));
+    // if user has previously already selected some payment methods and clicked Apply
+    const has_already_selected_payment_methods =
+        buy_sell_store.selected_payment_method_value?.length &&
+        (selected_methods.length !== buy_sell_store.selected_payment_method_value.length ||
+            diff(selected_methods, buy_sell_store.selected_payment_method_value).length > 0);
+    // if user is selecting payment methods for the first time and has selected some payment methods
+    const has_recently_selected_payment_methods =
+        buy_sell_store.selected_payment_method_value?.length === 0 && selected_methods.length > 0;
+    // if user has previously selected all payment methods
+    const has_selected_all_payment_methods =
```

Comment:
Added another check where if the user has selected all payment methods previously, then just disable the `Apply` button but do not disable `Clear` button